### PR TITLE
Add collapse buttons to groups on demand

### PR DESCRIPTION
**Pull request by swalterfub**
This addresses issue #15 (collapse layer groups)

### CODE REVIEW

File: examples/collapse-groups.html [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8" />
+    <title>OpenLayers 3 - LayerSwitcher</title>
+    <meta name="viewport" content="initial-scale=1.0, user-scalable=no, width=device-width">
+    <link rel="stylesheet" href="https://openlayers.org/en/v5.1.3/css/ol.css" />
+    <link rel="stylesheet" href="../src/ol-layerswitcher.css" />
+    <link rel="stylesheet" href="layerswitcher.css" />
+    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.3.1/css/all.css" integrity="sha384-mzrmE5qonljUremFsqc01SB46JvROS7bZs3IO2EmfFsd15uHvIt+Y8vEf7N7fWAU" crossorigin="anonymous">
```
Comment (walkermatt) [inline examples/collapse-groups.html]: I'd prefer it if we could avoid adding a dependency on fontawesome; the arrows could be either pure CSS (along the lines of https://css-tricks.com/the-shapes-of-css/) or base64 encoded images in the stylesheet.

File: src/ol-layerswitcher.js [hunk 2]
```diff
@@ -183,9 +183,33 @@ export default class LayerSwitcher extends Control {
         if (lyr.getLayers && !lyr.get('combine')) {
 
             li.className = 'group';
+            
+            //Group folding
+            var ulStyle, ulHeight;
+            if (lyr.get('unfolded')) {
+              lyr.id = lyrId;
+              var fold = document.createElement('i');
```
Comment (walkermatt) [inline src/ol-layerswitcher.js]: If we use CSS for the arrow then we might be able to remove the need for an extra element and instead style the group title and make the group title clickable?

**Comment by walkermatt** [comment]
I wonder if the Group property should be `fold` with values `open` or `close`. A Group with either no `fold` property or `fold: false` would not be foldable.

**Comment by swalterfub** [comment]
Yes good point!

**Comment by swalterfub** [comment]
I have incorporated the changes as proposed by @walkermatt but I can't get it looking how it's supposed to do. Maybe some CSS wizard out there who could have a look?

**Comment by umbe1987** [comment]
@swalterfub Didn't have a chance to test myself, but this would work nicely I think: https://www.w3schools.com/howto/howto_css_arrows.asp.

**Comment by swalterfub** [comment]
Nice @umbe1987 ! There is a new solution in my repo... no dependencies. (i still think the font-awsome solutions is a bit more fancy, but I can totally live with this one. Minimal-Style... Also now you can click on the layer label to (un-)collapse...

**Comment by umbe1987** [comment]
@swalterfub that is sweet! Thanks for providing this functionality ;)

**Comment by walkermatt** [comment]
@swalterfub this is looking good. I've a few changes I'd like to make, can you add me as a collaborator on your fork so I can commit/ push to your branch?

**Comment by walkermatt** [comment]
Current WIP can be tested here: [swalterfub/ol-layerswitcher/master/examples/collapse-groups.html](https://rawgit.com/swalterfub/ol-layerswitcher/master/examples/collapse-groups.html)

**Comment by walkermatt** [comment]
@swalterfub, @umbe1987 I think I'm happy to merge/ release, anyone spot anything outstanding?

**Comment by umbe1987** [comment]
> @swalterfub, @umbe1987 I think I'm happy to merge/ release, anyone spot anything outstanding?

@walkermatt I think everything is super nice!
@swalterfub @walkermatt thank you both for this neat introduction :collision: :collision: :collision:

**Comment by swalterfub** [comment]
Fine from my side as well! Thanks!

**Comment by kgiil** [comment]
The classes close and open conflicts with classes in Bootstrap. If you change class names, i.e. closed and opened, it works well even when using bootstrap.

**Comment by walkermatt** [comment]
@kgiil thanks for the heads up, I've just tested the collapse-groups example with Bootstrap and the `close` class does mess our styles up :man_shrugging: 

To avoid clashes we're going to have to be more specific, I'll give it some thought.

**Comment by umbe1987** [comment]
@walkermatt I saw your commit and I agree it makes perfect sense to prefix `layer-switcher` classes with a identifiable/"less-clashing" name. Any chance this PR will get merged anytime soon? Cheers.

**Comment by walkermatt** [comment]
Merged in a1061301dae8864d1a3ba058e7a8ffed32fd9c19 

@swalterfub thanks for your work on this and thanks to @umbe1987 and @kgiil for your input :tada:

**Comment by walkermatt** [comment]
And released to npm as [v3.2.0](https://www.npmjs.com/package/ol-layerswitcher/v/3.2.0)

**Pull request closed without merging**